coffea/processor/dataframe.py

```diff
@@ -26,15 +23,23 @@ class LazyDataFrame(MutableMapping):
         flatten : bool
             Remove jagged structure from columns read
     """
-    def __init__(self, tree, entrystart=None, entrystop=None, preload_items=None, flatten=False):
-        import uproot
+
+    def __init__(
+        self, tree, entrystart=None, entrystop=None, preload_items=None, flatten=False
+    ):
         self._tree = tree
         self._flatten = flatten
-        self._branchargs = {'awkwardlib': awkward, 'flatten': flatten}
-        entrystart, entrystop = uproot.tree._normalize_entrystartstop(tree.numentries, entrystart, entrystop)
-        self._branchargs['entrystart'] = entrystart
-        self._branchargs['entrystop'] = entrystop
-        self._available = {k.decode('ascii') for k in self._tree.keys()}
+        self._branchargs = {
+            "decompression_executor": uproot4.source.futures.TrivialExecutor(),
+            "interpretation_executor": uproot4.source.futures.TrivialExecutor(),
```

Comment:
What do these executors do which is like the former 'awkwardlib' and 'flatten' options?

Reply:
These are more or less independent issues. This forces uproot to be single-threaded in this context (which we want since the coffea processing is already multi-thread and we don't want to accidentally occupy more CPU than the user requests)

Reply:
Okay, and there is somewhere else in the code where the `flatten` argument is passed or is it unnecessary becuase you're asking all processors to be able to work with jaggedarrays?

Reply:
Indeed I marked `flatten` as deprecated: https://github.com/CoffeaTeam/coffea/pull/350/files#diff-2910e237aa690c4f123d26b3749619c3R1083 but it is still applied by the following code: https://github.com/CoffeaTeam/coffea/pull/350/files#diff-6a09dd7e5c8e03cd6def2d8ff9452ce6R57-R58

Reply:
Okay, fair enough.